### ISSUE TITLE
Chorus.VcsDrivers.Mercurial.UnrelatedRepositoryErrorException

### ISSUE THREAD

**Issue by megahirt** [sillsdev/LfMerge#183]
**shu-flex**

Chorus.VcsDrivers.Mercurial.UnrelatedRepositoryErrorException: The repository(a.k.a 'project' or 'collection') that you tried to synchronize with has the same name as yours, but it does not have the same heritage, so it cannot be synchronized. In order to Send/Receive projects, you have to start with a single project/collection, then copy that around. Don't feel bad if this is confusing, just ask for some technical help.
  at Chorus.VcsDrivers.Mercurial.HgRepository.PullFromTarget (System.String targetLabel, System.String targetUri) [0x00106] in <d98940b1597342908bfb88b257fca6f5>:0
  at Chorus.VcsDrivers.Mercurial.HgNormalTransport.Pull () [0x00000] in <d98940b1597342908bfb88b257fca6f5>:0
  at Chorus.VcsDrivers.Mercurial.HgRepository.Pull (Chorus.VcsDrivers.RepositoryAddress source, System.String targetUri) [0x00050] in <d98940b1597342908bfb88b257fca6f5>:0
  at LfMergeBridge.LanguageForgeSendReceiveActionHandler.LibTriboroughBridgeChorusPlugin.Infrastructure.IBridgeActionTypeHandler.StartWorking (SIL.Progress.IProgress progress, System.Collections.Generic.Dictionary`2[TKey,TValue] options, System.String& somethingForClient) [0x001cd] in <dbee31384bce49b1bd2ffaab93484008>:0
  at LfMergeBridge.LfMergeBridge.Execute (System.String actionType, SIL.Progress.IProgress progress, System.Collections.Generic.Dictionary`2[TKey,TValue] options, System.String& somethingForClient) [0x000df] in <dbee31384bce49b1bd2ffaab93484008>:0
  at LfMerge.Core.Actions.SynchronizeAction.DoRun (LfMerge.Core.ILfProject project) [0x0013c] in /build/lfmerge-7000072-9dfRZq/lfmerge-7000072-2.0.90.52/src/LfMerge.Core/Actions/SynchronizeAction.cs:98
  at LfMerge.Core.Actions.Action.Run (LfMerge.Core.ILfProject project) [0x0010d] in /build/lfmerge-7000072-9dfRZq/lfmerge-7000072-2.0.90.52/src/LfMerge.Core/Actions/Action.cs:142
  at LfMerge.Program.RunAction (System.String projectCode, LfMerge.Core.Actions.ActionNames currentAction) [0x000ed] in /build/lfmerge-7000072-9dfRZq/lfmerge-7000072-2.0.90.52/src/LfMerge/Program.cs:128

**Comment by josephmyers** [comment]
Stacktrace is busted, but it was easy to find the line in Bridge that caused the error (line 94 in LanguageForgeSendReceiveActionHandler on develop). As for the actual error, it seems somewhat self-explanatory and comes from here [https://github.com/sillsdev/chorus/blob/master/src/LibChorus/VcsDrivers/Mercurial/HgRepository.cs](url)

**Comment by megahirt** [comment]
Yes.  More specifically, the Language Depo repo has been "reset" by tech
support i.e. folder deleted and new repo created in its place, followed by
a FLEx "push" to repopulate the repo.  Whenever we get an "unrelated repo"
exception in LFMerge, we should simply delete our repo copy and then clone
a fresh one from LD, then carry on the original operation.
